test/testdata/library_simple.proto

```diff
@@ -141,3 +147,35 @@ message ListBooksResponse {
   // page of results.
   string next_page_token = 2;
 }
+
+
+// Request message for LibraryService.ListBooks.
+message ListShelvesRequest {
+  // The name of the project whose shelves we'd like to list.
+  string name = 1 [
+    (google.api.resource_reference).child_type = "library.googleapis.com/Shelf"
```

Comment:
This new ListShelves method is introduced to test that we create a ProjectName from this child_type.